src/python/pants/base/exception_sink.py

```diff
@@ -252,6 +247,32 @@ def _recapture_fatal_error_log_streams(cls, new_log_location):
 
     return (pid_specific_error_stream, shared_error_stream)
 
+  @classmethod
+  def reset_graceful_signal_handler(cls, graceful_signal_handler):
+    """
+    Class state:
+    - Overwrites `cls._graceful_signal_handler`.
+    OS state:
+    - Overwrites signal handlers for SIGINT, SIGQUIT, and SIGTERM.
+
+    :returns: a dict (int -> func) mapping signal numbers to handlers which can be used to restore
+              the previous handlers for the affected signals later.
+    """
+    assert(isinstance(graceful_signal_handler, GracefulSignalHandler))
+    # NB: Modify process-global state!
+    replaced_previous_handlers_by_signal = {
+      signal.SIGINT: signal.signal(signal.SIGINT, graceful_signal_handler.handle_sigint),
+      signal.SIGQUIT: signal.signal(signal.SIGQUIT, graceful_signal_handler.handle_sigquit),
+      signal.SIGTERM: signal.signal(signal.SIGTERM, graceful_signal_handler.handle_sigterm),
+    }
+    # Retry any system calls interrupted by any of the signals we just installed handlers for
+    # (instead of having them raise EINTR). siginterrupt(3) says this is the default behavior on
+    # Linux.
```

Comment:
After reading again, it looks like `False` is also the default on osx, and also what we want (since we don't have any checks for `EINTR` anywhere in the codebase) -- I've left a comment to this effect.